core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -200,7 +211,110 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
   }
 
   @Override
-  public void close() throws IOException {
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    if (meta.size() > 0) {
+      LOG.warn("Hadoop Catalog not support metadata {} on namespace: {}", meta, namespace);
+    }
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+
+    try {
+      if (isNamespace(fs, nsPath)) {
+        throw new org.apache.iceberg.exceptions.AlreadyExistsException("Namespace '%s' already exists!",
+            namespace);
+      }
+      fs.mkdirs(nsPath);
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Create namespace failed: %s", namespace);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    List<Namespace> namespaceList = new ArrayList<>();
+    String[] namespaces;
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+    try {
+      if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
+        throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+      }
+
+      List<String> pathList =  Stream.of(fs.listStatus(nsPath)).map(FileStatus::getPath)
+          .filter(path -> isNamespace(fs, path)).map(Path::getName).collect(Collectors.toList());
+
+      for (String path : pathList) {
```

Comment:
Looks like you accidentally missed my earlier comment:

> Why is this logic not part of the stream above?

I don't see a reason not to include this in the stream operation. So let's refactor to add it there, or help me understand why it's a good idea to do it this way.